integrations/gradle-bloop/src/main/scala/bloop/integrations/gradle/model/BloopConverter.scala

```diff
@@ -417,20 +469,45 @@ final class BloopConverter(parameters: BloopParameters) {
     projectDependencies.exists(dep => isProjectDependency(dep, resolvedArtifact))
   }
 
-  private def artifactToConfigModule(artifact: ResolvedArtifact): Config.Module = {
+  private def artifactToConfigModule(
+      artifact: ResolvedArtifact,
+      project: Project): Config.Module = {
+
+    val resolutionResult = project
+      .getDependencies()
+      .createArtifactResolutionQuery()
+      .forComponents(artifact.getId.getComponentIdentifier)
+      .withArtifacts(classOf[JvmLibrary], classOf[SourcesArtifact], classOf[JavadocArtifact])
+      .execute()
+
+    val name = artifact.getModuleVersion().getId().getName()
+    val resolvedSourcesDependencies =
+      resolutionResult
+        .getResolvedComponents()
+        .asScala
+        .flatMap(_.getArtifacts(classOf[SourcesArtifact]).asScala)
+        .collect {
+          case artifact: ResolvedArtifactResult =>
+            Config.Artifact(
+              name = name,
+              classifier = Option("sources"),
+              checksum = None,
+              path = artifact.getFile.toPath
+            )
+        }
+        .toList
+
     Config.Module(
-      organization = artifact.getModuleVersion.getId.getGroup,
-      name = artifact.getName,
-      version = artifact.getModuleVersion.getId.getVersion,
+      organization = artifact.getModuleVersion().getId().getGroup(),
```

Comment:
Why?

Reply:
I think it looks better that you actually know it's a function